tsdb/blockwriter.go

```diff
@@ -52,26 +52,29 @@ func durationToMillis(t time.Duration) int64 {
 // contains anything at all. It is the caller's responsibility to
 // ensure that the resulting blocks do not overlap etc.
 // Writer ensures the block flush is atomic (via rename).
-func NewBlockWriter(logger log.Logger, dir string, blockSize int64) *BlockWriter {
-	return &BlockWriter{
+func NewBlockWriter(logger log.Logger, dir string, blockSize int64) (*BlockWriter, error) {
+	w := &BlockWriter{
 		logger:         logger,
 		destinationDir: dir,
 		blockSize:      blockSize,
 	}
+	err := w.initHead()
+	if err != nil {
```

Comment:
We have few other nits to fix, so just another one but not a blocker :)
```suggestion
	if err := w.initHead(); err != nil {
```